### ISSUE TITLE
It lags out my VS installation

### ISSUE THREAD

**Issue by lucaciandrei** [icsharpcode/RefactoringEssentials#67]
I got to a point where this extension would keep my CPU @ 25-50% not doing anything with my VS, and also lagging it out. Is it a known bug, or not?
After disabling the extension, everything work well again (it's a pure 2015 installation, without any extension or whatnot).

**Comment by christophwille** [comment]
I take a project is loaded - can you share a repro / repro with a public project?

**Comment by lucaciandrei** [comment]
Yes, a project indeed is loaded, a somewhat large one. The repo unfortunately is private, and I cannot share it. I can definitely say the solution has over 100k LoC. Does the size of the solution matter in the performance of the extension?

**Comment by Rpinski** [comment]
Maybe, this is something we have to try to reproduce. Do you see any analyzer warning (RECSxxxx) appearing especially often? Not only the size might have an impact, but also how the project's code looks like.

**Comment by jumpinjackie** [comment]
I have a codebase which might be a suitable test case for you

svn clone https://svn.osgeo.org/mapguide/trunk/Tools/Maestro/
at revision 8704
and open Maestro\Maestro_All.sln
It's a 36-project solution

rev 8704 was the state of the code before I went on the great refactoring essentials cleanup job on it. Many VS crashes and crazy memory usage occurrences were had during my cleanup adventure.

**Comment by Rpinski** [comment]
Thank you for the hint! I see > 6000 warnings there, many of them from RE analyzers. I'll have a deeper look on that in the next days.

**Comment by Rpinski** [comment]
What I've seen with that Maestro project until now is that VS hangs completely when moving the mouse over a project (or the solution) name in Solution Explorer and scrolling through the list of analyzer findings appearing in a popup. I had to kill the process finally.

Apart from that I haven't seen any VS crashes, applying fixes (even batch fixes) also seemed to work.
@jumpinjackie Can you give an example of an applied fix or similar causing crashes or other weird behavior?

**Comment by christophwille** [comment]
If it is on IDE side, we might consider opening up an issue in the Roslyn repo, area IDE.

**Comment by lucaciandrei** [comment]
We'll have to take into consideration the fact that maybe it's not entirely VS's fault.
@Rpinski, were you using RE when you were navigating the projects. If you were, have you tried disabling RE?

**Comment by Rpinski** [comment]
What exactly do you mean by "navigating the projects"? If a project has some analyzer warnings, it's underlined in Solution Explorer and a popup is shown on mouse hovering, where you see all warnings in this project. When scrolling through this list, at some point VS will simply stop responding.

**Comment by Rpinski** [comment]
Just noticed that the popups in Solution Explorer, which I have described, seem to be a feature of "Productivity Power Tools" extension. That's not regular VS.
